src/runtime/dict.cpp

```diff
@@ -900,6 +900,136 @@ static int dict_print(PyObject* mp, FILE* fp, int flags) noexcept {
     return 0;
 }
 
+static Box* characterize(BoxedDict *a, BoxedDict *b, Box **pval) noexcept
+{
+    Box *akey = NULL; /* smallest key in a s.t. a[akey] != b[akey] */
+    Box *aval = NULL; /* a[akey] */
+    int cmp;
+
+    for (const auto& p : a->d) {
+        Box *thiskey, *thisaval, *thisbval;
+        if (p.first.value == NULL)
+            continue;
+
+        thiskey = p.first.value;
+        Py_INCREF(thiskey);  /* keep alive across compares */
+        if (akey != NULL) {
+            cmp = PyObject_RichCompareBool(akey, thiskey, Py_LT);
+
+            if (cmp < 0) {
+                Py_DECREF(thiskey);
+                goto Fail;
+            }
+            if (cmp > 0 || p.second == NULL){
+                /* Not the *smallest* a key; or maybe it is
+                 * but the compare shrunk the dict so we can't
+                 * find its associated value anymore; or
+                 * maybe it is but the compare deleted the
+                 * a[thiskey] entry.
+                 */
+                Py_DECREF(thiskey);
+                continue;
+            }
+        }
+
+        /* Compare a[thiskey] to b[thiskey]; cmp <- true iff equal. */
+        thisaval = p.second;
+        assert(thisaval);
+        Py_INCREF(thisaval);   /* keep alive */
+
+        BoxedDict::DictMap::iterator it;
+        try {
+            it = b->d.find(thiskey);
+            thisbval = it->second;
+        } catch (ExcInfo e) {
+            setCAPIException(e);
```

Comment:
Yes, It seems a problem because of thisbval uninitialized.
The compiler no error when I run make python_release, but can not build successed in travis_ci
failed log can't open in travis ci yesterday, so I  did not realize that.

And I will use clang-format to match yours rule, to fix that, add a goto Fail is a good advice, put a new PR after this, thanks.